### ISSUE TITLE
Unable to add tags on file

### ISSUE THREAD

**Issue by barrykeenan** [chrometoasters/silverstripe-advanced-taxonomies#30]
Goal is to extend a file edit form to add tags to a file.

When CMS user attempts to add a tag, the `TreeMultiselectField` gives this error:

```
Couldn't run query: SELECT DISTINCT "ParentID", COUNT(DISTINCT "AT_BaseObject"."ID") AS "NumChildren" FROM "AT_BaseObject" WHERE ("AT_BaseObject"."ParentID" "AT_BaseObject"."ID") GROUP BY "ParentID" Unknown column 'ParentID' in 'field list’
```

Partial trace:

```
SilverStripe\ORM\Hierarchy\Hierarchy::prepopulate_numchildren_cache(Chrometoaster\AdvancedTaxonomies\Models\TaxonomyTerm)
TreeDropdownField.php:463
```

Some options might be to move ParentID from TaxonomyTerm to BaseObject, or set a custom setNumChildrenMethod() to avoid prepopulate_numchildren_cache being called.

**Comment by michalkleiner** [comment]
ParentID shouldn't be moved, BaseObject and BaseTerm are not hierarchical.

**Comment by michalkleiner** [comment]
Related https://github.com/silverstripe/silverstripe-framework/issues/6127

**Comment by michalkleiner** [comment]
Not an issue in this module, will be fixed when https://github.com/silverstripe/silverstripe-framework/pull/10077 is resolved.

**Comment by michalkleiner** [comment]
Can confirm https://github.com/silverstripe/silverstripe-framework/pull/10077 fix resolves the issue.

**Comment by michalkleiner** [comment]
Related PR confirmed to fix this issue and merged into the 4.9 framework branch.

Before that gets into a stable release, it's possible to use [dev-pulls/6127-hierarchy-query-class-fix-4.8 as 4.8.1](https://github.com/chrometoasters/silverstripe-framework/tree/pulls/6127-hierarchy-query-class-fix-4.8) replacement for silverstripe/framework in the 4.8 line.
